### PR TITLE
Remove apkbuilder from third_party

### PR DESCRIPTION
**Pull request by meteorcloudy**

### CODE REVIEW

File: third_party/java/apkbuilder/BUILD.tools [hunk 1]
```diff
@@ -1,15 +0,0 @@
-# The build file for apkbuilder when it's in bazel's embedded tools repo.
-
-licenses(["notice"])  # Apache 2.0
-
-java_binary(
```
Comment (meisterT) [inline third_party/java/apkbuilder/BUILD.tools:5]: Did you check whether that file is referenced from any other place?

Comment (meteorcloudy) [reply third_party/java/apkbuilder/BUILD.tools:5]: I couldn't find any reference to [third_party/java/apkbuilder](https://github.com/bazelbuild/bazel/search?q=third_party%2Fjava%2Fapkbuilder&unscoped_q=third_party%2Fjava%2Fapkbuilder) except in third_party/BUILD

**Comment by jin** [comment]
This is still used for [--[no]use_singlejar_apkbuilder](https://docs.bazel.build/versions/master/command-line-reference.html#flag--use_singlejar_apkbuilder). If disabled, the legacy apkbuilder is used.'

That said, disabling the flag on even the most basic android build throws a NPE:

```
$ git clone https://github.com/bazelbuild/rules_jvm_external
$ cd rules_jvm_external/examples/android_instrumentation_test
$ bazel cquery ... --nouse_singlejar_apkbuilder
FAILED: Build did NOT complete successfully (0 packages loaded, 7942 targets configured)
Internal error thrown during build. Printing stack trace: java.lang.RuntimeException: Unrecoverable error while evaluating node '//src/main:greeter_app BuildConfigurationValue.Key[85500f71598cf9f881e7c0745b0a004bd764ac8a6c85b5f8c355a4d4fbd6528a] false' (requested by nodes '//src/test:greeter_test_app BuildConfigurationValue.Key[85500f71598cf9f881e7c0745b0a004bd764ac8a6c85b5f8c355a4d4fbd6528a] false')
	at com.google.devtools.build.skyframe.AbstractParallelEvaluator$Evaluate.run(AbstractParallelEvaluator.java:515)
	at com.google.devtools.build.lib.concurrent.AbstractQueueVisitor$WrappedRunnable.run(AbstractQueueVisitor.java:399)
	at java.base/java.util.concurrent.ForkJoinTask$AdaptedRunnableAction.exec(Unknown Source)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(Unknown Source)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(Unknown Source)
	at java.base/java.util.concurrent.ForkJoinPool.scan(Unknown Source)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(Unknown Source)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(Unknown Source)
Caused by: java.lang.NullPointerException
	at com.google.devtools.build.lib.analysis.actions.SpawnAction$Builder.setExecutable(SpawnAction.java:1022)
	at com.google.devtools.build.lib.rules.android.ApkActionsBuilder.legacyBuildApk(ApkActionsBuilder.java:182)
	at com.google.devtools.build.lib.rules.android.ApkActionsBuilder.registerActions(ApkActionsBuilder.java:158)
	at com.google.devtools.build.lib.rules.android.AndroidBinary.createAndroidBinary(AndroidBinary.java:568)
	at com.google.devtools.build.lib.rules.android.AndroidBinary.init(AndroidBinary.java:363)
	at com.google.devtools.build.lib.rules.android.AndroidBinary.create(AndroidBinary.java:118)
	at com.google.devtools.build.lib.rules.android.AndroidBinary.create(AndroidBinary.java:96)
	at com.google.devtools.build.lib.analysis.ConfiguredTargetFactory.createRule(ConfiguredTargetFactory.java:484)
	at com.google.devtools.build.lib.analysis.ConfiguredTargetFactory.createConfiguredTarget(ConfiguredTargetFactory.java:190)
	at com.google.devtools.build.lib.skyframe.SkyframeBuildView.createConfiguredTarget(SkyframeBuildView.java:888)
	at com.google.devtools.build.lib.skyframe.ConfiguredTargetFunction.createConfiguredTarget(ConfiguredTargetFunction.java:899)
	at com.google.devtools.build.lib.skyframe.ConfiguredTargetFunction.compute(ConfiguredTargetFunction.java:338)
	at com.google.devtools.build.skyframe.AbstractParallelEvaluator$Evaluate.run(AbstractParallelEvaluator.java:438)
	... 7 more
```

so we should actually just tear down the legacy ApkBuilder code and deps, which includes this PR.

**Comment by meteorcloudy** [comment]
> so we should actually just tear down the legacy ApkBuilder code and deps, which includes this PR.

That would be great!

But I don't see Bazel package this apkbuilder as embedded tool, I guess deleting this won't make things worse?

**Comment by meteorcloudy** [comment]
Or should we add `//third_party/java/apkbuilder:embedded_tools` to embedded tool, so that `--nouse_singlejar_apkbuilder` can work?

**Comment by jin** [comment]
I'm hesitant to reintroduce the legacy behavior and increase the size of the embedded tools for a behavior that has been broken for a long time, and nobody has complained about it :)

That said, I have to take back the idea of tearing down the entire legacy apkbuilder infrastructure, because there are still many references to the --nouse_singlejar_apkbuilder flag in google3. 

Let's just remove this from the Bazel codebase. I don't expect us to reintroduce functionality with the legacy apkbuilder in the native android rules. cc @ahumesky

**Comment by meteorcloudy** [comment]
Sounds good!

**Comment by meteorcloudy** [comment]
Merged at 3b5bb47d2743c3521c6e79f972a9ea638935c332

**Pull request closed without merging**